### DISCUSSION TITLE
How to repeat the action on the last list item?

### DISCUSSION THREAD

**Discussion by jackieli-tes** [neoclide/coc.nvim#3228] [Q&A] [answered]
I'm using a list called coc-tasks: https://github.com/voldikss/coc-tasks. 

My workflow:

- I bring up the list using `:CocList tasks`
- fuzzy search `build` for a build task
- enter to execute the item
- I do some editing
- now I want to run the same `build` task again
- I do `:CocListResume`
- I press enter

Ideally I'd like to do the last two steps in one if there is a `:CocListLastAction`? Or is there something like this already?

**Comment by chemzqm** [comment, marked as answer]
You can't, there's only repeat support for CocCommand `:h <plug>(coc-command-repeat)`, so it has to be registered as command.

**Comment by chemzqm** [comment]
My suggestion is write some script to run your build on save, so no need to run the command yourself, my esbuild extension:

```ts
// run esbuild on file change
const {spawn} = require('child_process')
const {workspace, Mutex, window} = require('coc.nvim')
const path = require('path')
const fs = require('fs')
const debounce = require('debounce')
const languageIds = ['javascript', 'typescript', 'javascriptreact', 'typescriptreact']

exports.activate = async context => {
  let mutex = new Mutex()
  let statusItem = window.createStatusBarItem(0, {progress: true})
  context.subscriptions.push(statusItem)
  async function runEsbuild(filepath) {
    let release = await mutex.acquire()
    let d = Date.now()
    let cp = spawn('node', ['esbuild.js'], {cwd: path.dirname(filepath)})
    statusItem.text = 'Running node esbuild.js'
    statusItem.show()
    cp.on('error', e => {
      window.showErrorMessage(`Error: ${e.message}`)
      statusItem.hide()
      release()
    })
    cp.stderr.on('data', chunk => {
      window.showErrorMessage(`Esbuild error: ${chunk.toString()}`)
    })
    cp.on('exit', (code) => {
      statusItem.hide()
      if (code == 0) {
        window.showMessage(`Build finished ${Date.now() - d}ms`, 'more')
      }
      release()
    })
  }

  let run = debounce(runEsbuild, 200)

  workspace.onDidSaveTextDocument(doc => {
    if (!languageIds.includes(doc.languageId)) return
    let cwd = workspace.cwd
    let filepath = path.join(cwd, 'esbuild.js')
    if (fs.existsSync(filepath)) {
      run(filepath)
    }
  }, null, context.subscriptions)
}
```

**Reply by jackieli-tes** [reply]
Yes, this is handy.

My use case is slightly different through:

I bring up my list and I select an item, then I invoke the action on that item. So I could select different actions and repeat that.

It's bit like intellij's `<Shift+F10>` or `Run -> Run ???` which runs last Run action.

So I can have one key to invoke different item for different projects.

**Reply by chemzqm** [reply]
The extension you're using could provide such kind of feature.

**Reply by jackieli-tes** [reply]
Yes, you're right.

Done it: https://github.com/voldikss/coc-tasks/pull/4/files
